### PR TITLE
docs: added mongoose connection status tutorial

### PR DESCRIPTION
**Pull request by IslandRhythms**
Should I also include an example? And if so, is there a folder in the repo where I could make a dummy run so I could show what a console.log of the call would return? Also I just made a second commit because I noticed in the stackoverflow comments that there is a fourth state that is not listed in the doc and I included that in the tutorial.

### CODE REVIEW

File: tutorials/mongoose/connection-status.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+['Mongoose has a readyState property in the Connection route of the API']('https://mongoosejs.com/docs/api/connection.html#connection_Connection-readyState') that is of type **property** and will return a number representing the current state of the connection, 0-4. These states are as follows:
```
Comment (vkarpov15) [inline tutorials/mongoose/connection-status.md:1]: Why are there single quotes around 'Mongoose has a ...' and the `https://mongoosejs.com` url?

Comment (IslandRhythms) [reply tutorials/mongoose/connection-status.md:1]: my bad on the single quotes, I was looking at the timestamps.md to see how to link to something and thought I saw single quotes in those sections. Taking them out now.

File: examples/mongoose.test.js [hunk 2]
```diff
@@ -1055,6 +1055,24 @@ describe('Mongoose', function() {
       // acquit:ignore:end
       // Connect to a MongoDB server running on 'localhost:27017' and use the
       // 'test' database.
+      console.log(mongoose.connection.readyState); //logs 0
```
Comment (vkarpov15) [inline examples/mongoose.test.js:1058]: I'd like these changes to be in a separate test. This test is already used as an example in another tutorial `connect.md`, and I'd rather not have these changes affect the other tutorial.

File: examples/mongoose.test.js [hunk 3]
```diff
@@ -1055,6 +1055,24 @@ describe('Mongoose', function() {
...
+      mongoose.connection.on('connecting', () =>{
```
Comment (vkarpov15) [inline examples/mongoose.test.js:1059]: Space between `=>` and `{`

**Review by vkarpov15 — CHANGES_REQUESTED** [review]

**Review by vkarpov15 — APPROVED** [review]

**Pull request merged**